src/CfgEditor/EdgeTPUCfgData.ts

```diff
@@ -0,0 +1,122 @@
+    return ini.stringify(this._edgeTpuConfig);
+  }
+
+  setWithConfig(cfg: any): void {
+    this._edgeTpuConfig = cfg;
+    this.resolveDuplicated();
+  }
+
+  setWithString(text: string): void {
+    this._edgeTpuConfig = ini.parse(text);
+    this.resolveDuplicated();
+  }
+
+  private resolveDuplicated(): void {
+    // NOTE 'intermediate_tensors' and a set of 'search_delegate' and 'delegate_search_step' cannot be duplicated.
+    //      Therefore, when only 'intermediate_tensors' and a set of 'search_delegate' and 'delegate_search_step' is used together, those three options will be deleted.
+    if (
+      this._edgeTpuConfig["edgetpu-compile"]["intermediate_tensors"] !==
+      undefined
+    ) {
+      if (
+        this._edgeTpuConfig["edgetpu-compile"]["search_delegate"] !== undefined
+      ) {
+        delete this._edgeTpuConfig["edgetpu-compile"]["intermediate_tensors"];
+        delete this._edgeTpuConfig["edgetpu-compile"]["search_delegate"];
+        if (
+          this._edgeTpuConfig["edgetpu-compile"]["delegate_search_step"] !==
+          undefined
+        ) {
+          delete this._edgeTpuConfig["edgetpu-compile"]["delegate_search_step"];
+        }
+      }
+    }
+  }
+
+  updateSectionWithKeyValue(section: string, key: string, value: string): void {
+    if (this._edgeTpuConfig[section] === undefined) {
+      this._edgeTpuConfig[section] = {};
+    }
+
+    if (this._edgeTpuConfig[section][key] === undefined) {
+      this._edgeTpuConfig[section][key] = "";
+    }
+    this._edgeTpuConfig[section][key] = value;
+    this.resolveDuplicated();
+  }
+
+  updateSectionWithValue(section: string, value: string): void {
+    // value should be encoded or stringfied
+    this._edgeTpuConfig[section] = ini.parse(value);
+    this.resolveDuplicated();
+  }
+
+  isSame(textStringified: string): boolean {
+    const iniDocument = ini.parse(textStringified);
+    for (const [sectionName, section] of Object.entries(this._edgeTpuConfig)) {
+      for (const [paramName, param] of Object.entries(section as any)) {
+        if (
+          iniDocument[sectionName] !== undefined &&
+          iniDocument[sectionName][paramName] === param
+        ) {
+          continue;
+        }
+        return false;
+      }
+    }
+    for (const [sectionName, section] of Object.entries(iniDocument)) {
+      for (const [paramName, param] of Object.entries(section as any)) {
+        if (
+          this._edgeTpuConfig[sectionName] !== undefined &&
+          this._edgeTpuConfig[sectionName][paramName] === param
+        ) {
+          continue;
+        }
+        return false;
+      }
+    }
+    return true;
+  }
+
+  sort(): void {
+    // cfg file is written along with the order of array elements
+    let sorted: any = {};
+    sections.forEach((section) => {
+      if (this._edgeTpuConfig[section] !== undefined) {
+        sorted[section] = this._edgeTpuConfig[section];
+      }
+    });
+    this.setWithConfig(sorted);
+  }
```

Comment:
I think isSame and sort function are same in CfgData and EdgeTPUCfgData both, maybe it can be shared.

Reply:
Let's change from interface to Abstract class to implement common member variables and functions!

Reply:
Cool, thanks!!

Reply:
I have done this issue with following commits [3d0c3e6](https://github.com/Samsung/ONE-vscode/pull/1673/commits/3d0c3e636ff19d617040161a5826dc0472c1d309) , [a44729c](https://github.com/Samsung/ONE-vscode/pull/1673/commits/a44729cff341648bdf56453625d1e67a08b6d8d5)